// @akashic-extension/akashic-label/lib/FragmentDrawInfo.js
window.gLocalAssetContainer["node_modules/@akashic-extension/akashic-label/lib/FragmentDrawInfo.js"] = function(g) { (function(exports, require, module, __filename, __dirname) {
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
/**
 * 行に含まれる文字列要素。
 */
var StringDrawInfo = /** @class */ (function () {
    function StringDrawInfo(text, width, glyphs) {
        this.text = text;
        this.width = width;
        this.glyphs = glyphs;
    }
    return StringDrawInfo;
}());
exports.StringDrawInfo = StringDrawInfo;
/**
 * 行に含まれるルビ要素。
 */
var RubyFragmentDrawInfo = /** @class */ (function () {
    function RubyFragmentDrawInfo(fragment, width, rbWidth, rtWidth, glyphs, rubyGlyphs) {
        this.text = fragment.text;
        this.fragment = fragment;
        this.width = width;
        this.rbWidth = rbWidth;
        this.rtWidth = rtWidth;
        this.glyphs = glyphs;
        this.rubyGlyphs = rubyGlyphs;
    }
    return RubyFragmentDrawInfo;
}());
exports.RubyFragmentDrawInfo = RubyFragmentDrawInfo;

})(g.module.exports, g.module.require, g.module, g.filename, g.dirname);
}
